packages/odyssey-react-mui/src/DataTable/DataTable.tsx

```diff
@@ -678,6 +678,13 @@ const DataTable = ({
     onChangeRowSelection?.(rowSelection);
   }, [rowSelection, onChangeRowSelection]);
 
+  const firstRow = pagination.pageSize * (pagination.pageIndex - 1) + 1;
+  let lastRow = firstRow + (pagination.pageSize - 1);
+  // If the last eligible row is greater than the number of total rows,
+  // show the number of total rows instead (ie, if we're showing rows
+  // 180-200 but there are only 190 rows, show 180-190 instead)
+  lastRow = totalRows && lastRow > totalRows ? totalRows : lastRow;
```

Comment:
This seems like it could be done in the Pagination component right?

Why'd we move it to `DataTable`?

I see, it's maintaining the page you're on and some other logic. Maybe `usePagination` would be better, so we could maintain the basic logic we're gonna copy-paste without having the view component handle it.

---

```tsx
lastRow = totalRows && lastRow > totalRows ? totalRows : lastRow;
```

Unless I'm misunderstanding, this code sets the type as `number | false`. Do we want to be explicit about checking `totalRows === 0`? I don't know what values to expect from this just by eyeballing it.


---

Side note: I don't like using `let`. We shouldn't be mutating values in JavaScript. You're doing it for readability, but then create a new variable. The compute cost is negligible, and the garbage collector will figure it out.

Still, this looks like something that could be in a separate function that could easily be unit tested. Something like `getLastPageValue(numberOfItems, lastRow)`? But I feel like I'm missing some context.

Wait a minute... Now I think I get it, why not this instead? (adjust the name as necessary):
```tsx
const lastPossibleRow = Math.max(numberOfItems, lastRow)
```

Reply:
The `usePagination` idea is fantastic, done!

---

`totalRows` could also be `undefined`, so checking `0` is not enough.

---

I simplified that code block, I haven't used `Math.min` but placed that condition inline in the return block.

---